controller/gateway/controller_reconciler_utils.go

```diff
@@ -702,7 +702,7 @@ func countAttachedRoutesForGatewayListener(ctx context.Context, g *gwtypes.Gatew
 							client.ObjectKeyFromObject(g), err,
 						)
 					}
-					count += int32(len(httpRoutes))
+					count += countAttachedHTTPRoutes(listener.Name, httpRoutes)
```

Comment:
I wonder why did this only come up now and not in the original PR that enabled `GatewayWithAttachedRoutes` https://github.com/Kong/gateway-operator/pull/286